### PR TITLE
Feature: User service added

### PR DESCRIPTION
**Pull request by pratikshabs-maveric**

### CODE REVIEW

File: src/main/java/com/maveric/userservice/controller/UserController.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+
+//@RequestMapping("/api/v1/users")
+//@RestController
+//public class UserController {
+//
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/controller/UserController.java]: please remove the commented code

File: src/main/java/com/maveric/userservice/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+//}
+@RequestMapping("/api/v1")
+@RestController
+public class UserController {
+
+    @Autowired
+    UserService userService;
+
+    @GetMapping("users")
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/controller/UserController.java:25]: move "users" to the base path as its common

File: src/main/java/com/maveric/userservice/dto/AppResponse.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.maveric.userservice.dto;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@AllArgsConstructor
+@Data
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/dto/AppResponse.java:9]: use @getter and @setter

File: src/main/java/com/maveric/userservice/dto/UserResponse.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+    private long id;
+    private String firstName;
+    private String lastName;
+    private String middleName;
+    private String phoneNumber;
+    private String email;
+    private String address;
+    private String dateOfBirth;
+    private String gender;
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/dto/UserResponse.java:20]: check the type of gender it should be enum

File: src/main/java/com/maveric/userservice/mapper/UserMapperImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                user.getGender()
+        );
+    }
+
+    @Override
+    public List<User> map(List<UserResponse> userResponses) {
+        List<User> list = new ArrayList<User>(userResponses.size());
+        for(UserResponse userResult:userResponses)
+        {
+            list.add(map(userResult));
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/mapper/UserMapperImpl.java:48]: use stream

File: src/main/java/com/maveric/userservice/enumeration/Gender.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package com.maveric.userservice.enumeration;
+
+public enum Gender {
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/enumeration/Gender.java:3]: where is this enum bieng used?

File: src/main/java/com/maveric/userservice/service/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    @Override
+    public String deleteUser(String userId) {
+        repository.deleteById(Long.valueOf(userId));
+        return "The user is deleted successfully.";
+    }
+
+    public UserResponse getUserDetailsByEmail(String email) {
+        User getEmailResult = repository.findByEmail(email);
```
Comment (sgorle10) [inline src/main/java/com/maveric/userservice/service/UserServiceImpl.java:60]: handle null case

**Review by sgorle10 — CHANGES_REQUESTED** [review]
check usage of annotations properly

File: src/main/java/com/maveric/userservice/controller/UserController.java (hunk 1 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/controller/UserController.java]: Done

File: src/main/java/com/maveric/userservice/controller/UserController.java (hunk 2 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/controller/UserController.java:25]: Done

File: src/main/java/com/maveric/userservice/dto/AppResponse.java (hunk 3 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/dto/AppResponse.java:9]: Unused file

File: src/main/java/com/maveric/userservice/enumeration/Gender.java (hunk 6 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/enumeration/Gender.java:3]: Fixed. Used in user.class file

File: src/main/java/com/maveric/userservice/dto/UserResponse.java (hunk 4 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/dto/UserResponse.java:20]: Fixed.

File: src/main/java/com/maveric/userservice/mapper/UserMapperImpl.java (hunk 5 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/mapper/UserMapperImpl.java:48]: Will you elaborate a little bit or share some article. @sgorle10

File: src/main/java/com/maveric/userservice/service/UserServiceImpl.java (hunk 7 above)
Comment (pratikshabs-maveric) [reply src/main/java/com/maveric/userservice/service/UserServiceImpl.java:60]: Fixed. Handled in UserServiceImpl.java file, where getEmailById() methods is written.

**Pull request closed without merging**